website/docs/r/appmesh_gateway_route.html.markdown

```diff
@@ -82,7 +83,6 @@ The `rewrite` object supports the following:
 
 * `hostname` - (Optional) Host name to rewrite.
 * `prefix` - (Optional) Specified beginning characters to rewrite.
-* `port` - (Optional) The port number to match from the request.
```

Comment:
The mistake is that there is suppose to be port available from match. It was just never implemented. In fact there are many options missing across the entire api of the appmesh components. You are supposed to be able to match everything that can be defined from one listener and then send it a completely different defined target listener. So these changes are also needed to be made to the regular appmesh route. 

AWS has a repo of appmesh examples. This example is impossible to implement using terraform right now. https://github.com/aws/aws-app-mesh-examples/tree/main/walkthroughs/howto-multiple-listeners/mesh

Reply:
Looks like you're right, I also found it in the API spec.
https://docs.aws.amazon.com/app-mesh/latest/APIReference/API_HttpGatewayRouteMatch.html

For regular Routes on Virtual Routers, is this not already implemented under `weighted_target`?